service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/ObservationMapper.java

```diff
@@ -95,6 +94,8 @@ public String map() {
                 .collect(Collectors.toList());
 
             var classCode = prepareClassCode(derivedObservations);
+            var compoundStatementId = idMapper.getOrNew(ResourceType.Observation, observationAssociatedWithSpecimen.getIdElement());
```

Comment:
this line will generate same id for each default observation if only there will be more default observations within on bundle. That will happen because idMapper store map of ids and its mappings and all default observations has the same id so mapped id will also be the same. Because the mapping of the ids must be consistent. 

We could try to add something random to the default observation id, that it would generate new id each time.